src/libcollections/vec.rs

```diff
@@ -1106,12 +1106,36 @@ impl<T: Clone> Vec<T> {
         let len = self.len();
 
         if new_len > len {
-            self.extend(repeat(value).take(new_len - len));
+            self.extend_elem(new_len - len, value);
```

Comment:
A comment suggesting this is the most performant way to do it right now will only get outdated.

A comment that explains this is the simplest method that does the job is just redundant I think.

Not sure what to say. "This used to use extend" isn't useful comment either I think.